frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx

```diff
@@ -0,0 +1,192 @@
+import * as React from 'react';
+import { match } from 'react-router';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import { Alert, ActionGroup, Button, Form, FormGroup } from '@patternfly/react-core';
+import {
+  NodeKind,
+  StorageClassResourceKind,
+  referenceForModel,
+  k8sCreate,
+  K8sResourceKind,
+} from '@console/internal/module/k8s';
+import { ListPage } from '@console/internal/components/factory';
+import { NodeModel } from '@console/internal/models';
+import {
+  withHandlePromise,
+  HandlePromiseProps,
+  history,
+  FieldLevelHelp,
+  ButtonBar,
+} from '@console/internal/components/utils';
+import { setFlag } from '@console/internal/actions/features';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { getName } from '@console/shared';
+import {
+  minSelectedNode,
+  storageClassTooltip,
+  defaultRequestSize,
+} from '../../../constants/ocs-install';
+import { NO_PROVISIONER } from '../../../constants';
+import { OCSServiceModel } from '../../../models';
+import { OCSStorageClassDropdown } from '../../modals/storage-class-dropdown';
+import AttachedDevicesNodeTable from './sc-node-list';
+import { PVsAvailableCapacity } from '../pvs-available-capacity';
+import { OCS_CONVERGED_FLAG, OCS_FLAG } from '../../../features';
+import { makeLabelNodesRequest } from '../create-form';
+import { lvsResource } from '../../../constants/resources';
+import { getOCSRequestData } from '../ocs-request-data';
+
+import '../ocs-install.scss';
+import './attached-devices.scss';
+
+const makeOCSRequest = (
+  selectedData: NodeKind[],
+  storageClass: StorageClassResourceKind,
+): Promise<any> => {
+  const promises = makeLabelNodesRequest(selectedData);
+  const scName = getName(storageClass);
+  const ocsObj = getOCSRequestData(scName, defaultRequestSize.BAREMETAL);
+
+  ocsObj.spec.monDataDirHostPath = '/var/lib/rook';
+  ocsObj.spec.storageDeviceSets[0].portable = false;
+
+  return Promise.all(promises).then(() => {
+    if (!scName) {
+      throw new Error('No StorageClass selected');
+    }
+    return k8sCreate(OCSServiceModel, ocsObj);
+  });
+};
+
+export const CreateOCS = withHandlePromise<CreateOCSProps & HandlePromiseProps>((props) => {
```

Comment:
There are things in here that are common for Attached and Internal Storage Mode. We need to move them to a commonplace and use them. We are basically repeating our code in two places inside the same plugin.